src/KlaviyoAPI.php

```diff
@@ -0,0 +1,368 @@
+    const SEND = 'send';
+    const TIMELINE = 'timeline';
+
+    /**
+     * Klaviyo API arguments
+     */
+    const PROFILES = 'profiles';
+    const COUNT = 'count';
+    const PAGE = 'page';
+    const SINCE = 'since';
+    const SORT = 'sort';
+
+    /**
+     * @var string
+     */
+    protected $private_key;
+
+    /**
+     * @var string
+     */
+    protected $public_key;
+
+    /**
+     * @var string
+     */
+    protected $client;
+
+    /**
+     * Constructor method for Base class.
+     *
+     * @param string $public_key Public key (account ID) for Klaviyo account
+     * @param string $private_key Private API key for Klaviyo account
+     * @param string $host Base URI for API requests. Can be overridden for testing.
+     */
+    // TODO: I don't think we can ever actually define a different host like this. Only calling this through __get in the Klaviyo class. Would like to change for local dev for example.
+    public function __construct( $public_key, $private_key )
+    {
+        $this->public_key = $public_key;
+        $this->private_key = $private_key;
+        $this->client = new Client(['base_uri' => self::BASE_URL]);
+    }
+    /**
+     * Make public API request
+     *
+     * @param $path Endpoint to call
+     * @param $options API params to add to request
+     */
+    protected function publicRequest( $path, $options )
+    {
+        // Public requests are always GET
+        return $this->request( self::HTTP_GET, $path, $options, true );
+    }
+
+    /**
+     * Make private v1 API request
+     *
+     * @param $path Endpoint to call
+     * @param $options API params to add to request
+     * @param string $method HTTP method for request
+     */
+    protected function v1Request( $path, $options = [], $method = self::HTTP_GET )
+    {
+        $path = self::API_V1 . $this->trimPath( $path );
+
+        return $this->request( $method, $path, $options, false, true );
+    }
+
+    /**
+     * Make private v2 API request
+     *
+     * @param $path Endpoint to call
+     * @param $options API params to add to request
+     * @param string $method HTTP method for request
+     */
+    protected function v2Request( $path, $options = [], $method = self::HTTP_GET )
+    {
+        $path = self::API_V2 . $this->trimPath( $path );
+
+        return $this->request( $method, $path, $options, false, false, true );
+    }
+
+    /**
+     * Make API request using HTTP client
+     *
+     * @param $path Endpoint to call
+     * @param $options API params to add to request
+     * @param string $method HTTP method for request
+     * @param bool $isPublic to determine if public request
+     * @param bool $isV1 to determine if V1 API request
+     * @param bool $isV2 to determine if V2 API request
+     */
+    private function request( $method, $path, $options, $isPublic = false, $isV1 = false, $isV2 = false )
+    {
+        $options = $this->prepareAuthentication( $options, $isPublic, $isV1, $isV2 );
```

Comment:
could we also add a user-agent to the headers of all these requests that takes the version number of the package?

similar to https://github.com/klaviyo/python-klaviyo/blob/master/klaviyo/api_helper.py#L216

Reply:
I'm versioning this one as 2.0.0, going with semver